### ISSUE TITLE
Ubuntu software update breaks functionality

### ISSUE THREAD

**Issue by perry-mitchell** [nylas/N1#1742]
I'm running N1 on 2 computers running Ubuntu 14.04. After installing N1 from a deb, updates will appear in software updates and will install successfully. N1 will lose certain icons (Mark as read, etc.), and the new-email functionality no longer works (both the button and ctrl-n).

Purging the installation and reinstalling from a deb does not fix the issue, and N1 is unusable from this point forward. I can't get it working again on either machine. Using the latest version (0.4.14). Not able to get the version from the menu, as it no longer exists either:

![image](https://cloud.githubusercontent.com/assets/3869469/13783817/40e889fa-eaa3-11e5-98d7-4893e3b6f682.png)

**Comment by perry-mitchell** [comment]
This may be related to #1543.

**Comment by perry-mitchell** [comment]
Closing and restarting N1 by just closing it does not fix the issue, as pointed out by @mbilker. You have to quit it properly. I had to close 4-5 different groups of processes before there was nothing matching "nylas" in my process list. After this, starting N1 seems to have produced a working instance.

Is there something that can be done to make this update procedure more clear? Updating while open compromises the stability of the application, when it shouldn't.

**Comment by mbilker** [comment]
@perry-mitchell It does compromise the integrity of the application. It is possible to add a script to the deb to check if N1 is running and halt the upgrade process or warn N1 to save all data and quit.

**Comment by bengotow** [comment]
Hmm I feel like we should be able to do this with a script but I'm unfamiliar with how updates work through `apt`. I think on Windows we fix this by unpacking the update into a directory next to the current one, and then the next time you quit N1 is moves the files into place and deletes the old directory. Not sure if something similar would make sense on linux.

**Comment by mbilker** [comment]
Unfortunately, the default installation path is only modifiable by `root`. The Windows installation is to `AppData`, which is user modifiable. I think the only way is to take advantage of the single instance ability and tell N1 to quit or notify the user to save all data and quit.

**Comment by perry-mitchell** [comment]
@mbilker I agree, though I am already prompted for a password when installing - is that not for root access?

Regardless, I like the idea of having a script run to close N1.

**Comment by mbilker** [comment]
@perry-mitchell 
TL;DR yes it needs your password to modify the filesystem as root.

Yes, `apt-get` or the `dpkg` package manager on Debian-derived Linux distributions install packages to the system. To modify anything on the system, elevated permissions are needed, which is why it needs your password. `sudo` is `(su)per-user (do)` and is the best way on Linux, OS X, and other *nix systems to run a command as the superuser or another user.

Now that I am thinking about this. `apt-get` runs as `root`, which is not the user's account and there is not a way to tell the user account that is running N1 except by checking the process list and killing the N1 process. The signal to the main Electron process could be handled to safely close down N1 and flush the database to disk.

**Comment by jackiehluo** [comment]
Closing this issue since it looks like it's (sort of) resolved and we haven't had much activity on this thread.
